[docs] Add Snackbars TypeScript Demos

sperry94 opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/next/CONTRIBUTING.md#submitting-a-pull-request).

Added TS demos for Snackbars components as a part of #14897

Also, rewrote the ConsecutiveSnackbars demo to use hooks to get around a proptypes typescript error with styles.

oliviertassinari commented on docs/src/pages/demos/snackbars/ConsecutiveSnackbars.js:
```diff
@@ -1,104 +1,117 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import { withStyles } from '@material-ui/core/styles';
+import React, { useState } from 'react';
```
As far as I know, there is no bundle size advantage. So, better improve the DX. It's easier to import once.

```suggestion
import React from 'react';
```

sperry94 commented:
> > Also, rewrote the ConsecutiveSnackbars demo to use hooks to get around a proptypes typescript error with styles.
> 
> Could you do this in a separate PR? You can look at the other demos how to get around `propTypes` errors.

@eps1lon @oliviertassinari I was able to find another demo with a pattern I could follow! I've updated `ConsecutiveSnackbar` to just convert to TS!

oliviertassinari commented:
@sperry94 Thanks
